external-tests/clone_repo.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -x
+
+if [ "$TESTS_REPO" == "" ]; then
+	echo "[ERROR]: No TESTS_REPO provided"
+	exit 1
+fi
+
+if [ "$TESTS_LOCATION" == "" ]; then
+	echo "[ERROR]: No TESTS_LOCATION provided"
+	exit 1
+fi
+
+# detect latest master hash of ocs-ci if we're not pinned to a specific commit hash
+if [ -z "$TESTS_TARGET_HASH" ]; then
+	echo "Using latest master branch commit for $TESTS_REPO"
+	TESTS_TARGET_HASH=$(git ls-remote "$TESTS_REPO" | grep refs/heads/master | cut -f 1)
+fi
+
+echo "$TESTS_REPO commit hash: $TESTS_TARGET_HASH"
+
+if ! [ -d "$TESTS_LOCATION" ]; then
+	DOWNLOAD_SRC=true
+elif ! [ "$(cat "$TESTS_LOCATION"/git-hash)" = "$TESTS_TARGET_HASH" ]; then
+	rm -rf TESTS_LOCATION
+	DOWNLOAD_SRC=true
+fi
+
+if [ $DOWNLOAD_SRC ]; then
+	echo "Cloning code from $TESTS_REPO using hash $TESTS_TARGET_HASH"
+        # shellcheck disable=SC2086
+    repo_name=$(basename $TESTS_REPO)
```

Comment:
spaces vs tabs. pick which ever one we've been consistently using for bash scripts.

Reply:
Oh, I did not even notice it locally. Will change it.

Reply:
Done